infrastructure/devicepathresolver/scsi_id_device_path_resolver.go

```diff
@@ -0,0 +1,108 @@
+package devicepathresolver
+
+import (
+	"regexp"
+	"strings"
+	"time"
+
+	boshsettings "github.com/cloudfoundry/bosh-agent/settings"
+	bosherr "github.com/cloudfoundry/bosh-utils/errors"
+	boshlog "github.com/cloudfoundry/bosh-utils/logger"
+	boshsys "github.com/cloudfoundry/bosh-utils/system"
+)
+
+// Resolves device path by performing a SCSI rescan then looking under
+// "/dev/disk/by-id/*uuid" where "uuid" is the cloud ID of the disk
+type scsiIDDevicePathResolver struct {
+	diskWaitTimeout time.Duration
+	fs              boshsys.FileSystem
+	logger          boshlog.Logger
+}
+
+func NewScsiIDDevicePathResolver(
+	diskWaitTimeout time.Duration,
+	fs boshsys.FileSystem,
+	logger boshlog.Logger,
+) DevicePathResolver {
+	return scsiIDDevicePathResolver{
+		diskWaitTimeout: diskWaitTimeout,
+		fs:              fs,
+		logger:          logger,
+	}
+}
+
+const logTag = "scsiIDresolver"
+
+func (idpr scsiIDDevicePathResolver) GetRealDevicePath(diskSettings boshsettings.DiskSettings) (string, bool, error) {
+	if diskSettings.ID == "" {
+		err := bosherr.Errorf("Disk ID is not set")
+		idpr.logger.Error(logTag, err.Error())
```

Comment:
agent doesnt log errors returned since they will be logged at the action handler level. you can also then inline err into the return statement.